File: inversao.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define false 0
#define true 1

typedef struct __fila__no{
    // nó da fila
    int data;
    struct __fila__no *prox;
} fila_no;

typedef struct __fila__type{
    // tipo / fila
    fila_no *inicio;
    fila_no *fim;
} fila_t;

int a,b;

int d[100010];

// variaveis
void busca_em_profundidade(int);
void pop(fila_t *);
int front(fila_t *);
_Bool vazio(fila_t *);
void push(fila_t *, int);
void cria_fila(fila_t *);

int main(int argc, char **argv){
    // principal
    int t;
    scanf("%d", &t);

    while(t--){
        memset(d, -1, sizeof(d));
        scanf("%d %d", &a, &b);
        busca_em_profundidade(a);
        printf("%d\n", d[b]);
    }
    return 0;
}

void busca_em_profundidade(int u){
    // realiza uma busca em profundidade
    fila_t fila; 
    cria_fila(&fila); // cria a fila
    push(&fila, u);

    d[u] = 0;

    while (!vazio(&fila)){
        // enquanto a fila nao estiver vazia
        int v = front(&fila);

        if (d[b] != -1)
            break;
        pop(&fila); // remove elemento

        int invert = 0, uu = v;
        while (uu){
            invert *= 10;
            invert += uu % 10;
            uu /= 10;
        }
        if (d[invert] == -1)
            d[invert] = d[v] + 1, push(&fila, invert);
        if (d[v + 1] == -1)
            d[v + 1] = d[v] + 1, push(&fila, v + 1);
    }
}

void cria_fila(fila_t *__fila){
    // cria a fila
    __fila->inicio = NULL; // posicao inicial
    __fila->fim = NULL;    // posicao final
}

void push(fila_t *__fila, int __data){
    // adiciona elemento na fila
    fila_no *tmp;  
    tmp = (fila_no *) malloc(sizeof(fila_no));

    if (__fila->inicio)
        __fila->fim->prox = tmp;
    else
        __fila->inicio = tmp;

    tmp->prox = NULL;
    __fila->fim = tmp;
    tmp->data = __data;

}

void pop(fila_t *__fila){
    // remove elemento da fila
    fila_no *tmp;

    if (__fila->inicio){
        // se a fila estiver no inicio
        if (__fila->inicio->prox){
            tmp = __fila->inicio;
            __fila->inicio = __fila->inicio->prox;
            free(tmp);              
        }
        else{
            tmp = __fila->inicio;
            __fila->inicio = __fila->fim = NULL;
            free(tmp);   
        }
    }
    else
        return;
}

_Bool vazio(fila_t *__fila){
    return __fila->inicio == NULL;
}

int front(fila_t *__fila){
    return __fila->inicio->data;
}

File: beecrowd/inversao.py
def cria_fila():
    # cria uma fila
    fila = {"inicio": None, "fim": None}
    return fila

def push(fila, data):
    # adiciona elemento na fila
    aux = {"data": data, "proximo": None}

    if fila["inicio"] is not None:
        fila["fim"]["proximo"] = aux
    else:
        fila["inicio"] = aux

    fila["fim"] = aux

def pop(fila):
    # retira elemento da fila
    if fila["inicio"] is not None:
        if fila["inicio"]["proximo"] is not None:
            aux = fila["inicio"]
            fila["inicio"] = fila["inicio"]["proximo"]
        else:
            aux = fila["inicio"]
            fila["inicio"] = fila["fim"] = None
        del aux

def ta_vazio(fila):
    # verifica se a fila tá vazia
    return fila["inicio"] is None

def front(fila):
    return fila["inicio"]["data"]

def busca_em_largura(u):
    # realiza uma busca em largura
    global tmp, b
    fila = cria_fila()
    push(fila, u)
    tmp[u] = 0
    while not ta_vazio(fila):
        v = front(fila)
        if tmp[b] != -1:
            break
        pop(fila)
        rev, uu = 0, v
        while uu > 0:
            rev *= 10
            rev += uu % 10
            uu //= 10
        if tmp[rev] == -1:
            tmp[rev] = tmp[v] + 1
            push(fila, rev)
        if tmp[v + 1] == -1:
            tmp[v + 1] = tmp[v] + 1
            push(fila, v + 1)

# começa o código aqui
tmp = [-1] * 10000  # limita a 10000
#print(tmp)
a, b = map(int, "808 909".split())

busca_em_largura(a)

print(tmp[b])